utils/dashboard-helper.ts

```diff
@@ -0,0 +1,128 @@
+import fetch from 'node-fetch';
+import { fetchPaginatedGHResults, filterOutTestFiles } from './lib/github-api-helpers';
+
+const guidRegex = /guid[`'"\) ]/;
+const entityGuidRegex = /entityGuid/;
+const entityGuidFieldRegex = /\"linkedEntityGuids\": (?:(?!null))/;
+const permissionsFieldRegex = /\"permissions\": /;
+const accountIdRegEx = /\"accountId\": (?:(?!0))/;
+
+export const checkLine = (line: string) => {
+  const warningsFound = [];
+
+  if (guidRegex.test(line)) {
+    warningsFound.push(`\"guid\" should not be used`);
+  }
+  if (entityGuidRegex.test(line)) {
+    warningsFound.push(`\"entityGuid\" should not be used`);
+  }
+  if (entityGuidFieldRegex.test(line)) {
+    warningsFound.push(`\"linkedEntityGuid\" must be set to null`);
+  }
+  if (permissionsFieldRegex.test(line)) {
+    warningsFound.push(`\"permissions\" field should not be used`);
+  }
+  if (accountIdRegEx.test(line)) {
+    warningsFound.push(`\"accountId\" must be zero`);
+  }
+
+  return warningsFound;
+};
```

Comment:
[nit] - Not a blocker at all, but I'm wondering if there is a way we can simplify this code to avoid so many `if` statements. In the chance we need to add more regEx in the future, having to add in the regex in 2 different places seems odd to me.

I put a suggestion below, but feel free to disregard it

```suggestion
const regexAndWarning = [
  [ /guid[`'"\) ]/, `\"guid\" should not be used` ],
  [ /entityGuid/, `\"entityGuid\" should not be used` ],
  [ /\"linkedEntityGuids\": (?:(?!null))/, `\"entityGuid\" should not be used`],
  [ /\"permissions\": /, `\"permissions\" field should not be used` ],
  [ /\"accountId\": (?:(?!0))/, `\"accountId\" must be zero` ],
]

export const checkLine = (line: string) => {
  const warningsFound = [];

  for const [regex, warning] of regexAndWarning {
    if (regex.test(line)) {
      warningsFound.push(warning)
    }
  }

  return warningsFound;
};
```

Reply:
@josephgregoryii I like it! I made these changes. Tests still pass, and it seems to be working as expected locally. I think it's ready for a re-review.